3rdparty/find_dependencies.cmake

```diff
@@ -853,6 +853,25 @@ if (BUILD_LIBREALSENSE)
     endif()
 endif()
 
+if(USE_SYSTEM_ZLIB)
+    open3d_find_package_3rdparty_library(3rdparty_zlib
+        PACKAGE ZLIB
+        TARGETS ZLIB::ZLIB
+    )
+    list(APPEND Open3D_3RDPARTY_PRIVATE_TARGETS_FROM_SYSTEM Open3D::3rdparty_zlib)
```

Comment:
I think You should set `USE_SYSTEM_ZLIB` to `OFF` if the package can't be found.